### DISCUSSION TITLE
setting playlist metadata (such as title) through libmpv

### DISCUSSION THREAD

**Discussion by brightening-eyes** [mpv-player/mpv#15194] [Q&A]
hello,
suppose I have lets say n items from which I add them with loadfile command with append argument, and those items do not have titles. how would I set title for them? (lets say from a customized string).
you know, playlist/N/title is not writable. (it should be a way possible to change the metadata).
thanks a lot

**Comment by guidocella** [comment]
loadfile lets you pass per-file options and you can pass `--force-media-title`.

**Comment by mrfragger** [comment]
Just pasting a video playlist from yt it'll populate the playlist with titles...but if one saves it it'll just save the URLs

I save the playlist then add media-title and reverse if necessary

#Playlist
d-p      run /bin/bash -c 'cd ~/datampv; [ ! -d playlists ] && mkdir playlists'; script-message save-playlist ~/datampv/playlists/    #! [Playlist] > [m3u] > Save .m3u playlist

d-k    run /bin/bash -c 'cd ~/datampv/playlists; [ ! -d titles ] && mkdir titles ; for f in *.m3u ; do yt-dlp --skip-download --flat-playlist --no-warnings -O "#EXTINF:,%(upload_date>%Y%m%d)s %(title)s %(duration>[%H:%M])s" -O webpage_url -a "$f" > titles/"$f" ; done; dt=$( date +%Y_%m_%d_%H_%M_%S); mkdir "$dt" && mv *.m3u "$dt"' ; show-text "Titles are being added to all yt .m3u playlists"#! [Playlist] > [m3u] > Add titles to all yt .m3u playlists

d-i    run "/bin/bash" "-c" 'cd ~/.config/mpv/extrastuff; ./m3u_yt_titlesreverse.sh' ; show-text "Titles in reverse are being added to all yt .m3u playlists" #! [Playlist] > [m3u] > Add titles in reverse to all yt .m3u playlists

and m3u_yt_titlesreverse.sh since can't get it to be in a run ..probably cuz of reversing it..so have to use it as a script

cd ~/datampv/playlists
[ ! -d titles_reversed ] && mkdir titles_reversed
for f in *.m3u 
    do cat $f | tail -r > ${f%.*}.m3u8
done
for f in *.m3u8
    do yt-dlp --skip-download --flat-playlist --no-warnings -O "#EXTINF:,%(upload_date>%Y%m%d)s %(title)s %(duration>[%H:%M])s %(uploader)s" -O webpage_url -a "$f" > titles_reversed/${f%.*}.m3u
done
dt=$( date +%Y_%m_%d_%H_%M_%S)
mkdir "$dt" && mv *.m3u *.m3u8 "$dt"

**Comment by brightening-eyes** [comment]
hello again,
so, mpv_command can accept --force-media-title as I can infer it.
but my question is, can I use it in mpv_command as an array of char pointers (which is simpler), or should I make nodes and populate those?
thanks.
